operations/node_management/Adding_a_Liquid-cooled_blade_to_a_System_Using_SAT.md

```diff
@@ -0,0 +1,334 @@
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#355:   name: 'x3000c0s19b1n0' -> 'x3000c0s19b1n0'
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#357:   address: '10.252.0.26@tcp99' -> '10.252.0.33@tcp99'
+   [Tue Jul 21 13:09:54 2020] DVS: merge_one#358:   Ignoring.
+   ```
+
+1. SSH to the node and check each DVS mount.
+
+   ```bash
+   nid# mount | grep dvs | head -1
+   ```
+
+   Example output:
+
+   ```text
+   /var/lib/cps-local/0dbb42538e05485de6f433a28c19e200 on /var/opt/cray/gpu/nvidia-squashfs-21.3 type dvs (ro,relatime,blksize=524288,statsfile=/sys/kernel/debug/dvs/mounts/1/stats,attrcache_timeout=14400,cache,nodatasync,noclosesync,retry,failover,userenv,noclusterfs,killprocess,noatomic,nodeferopens,no_distribute_create_ops,no_ro_cache,loadbalance,maxnodes=1,nnodes=6,nomagic,hash_on_nid,hash=modulo,nodefile=/sys/kernel/debug/dvs/mounts/1/nodenames,nodename=x3000c0s6b0n0:x3000c0s5b0n0:x3000c0s4b0n0:x3000c0s9b0n0:x3000c0s8b0n0:x3000c0s7b0n0)
+   ```
+
+#### Check the HSN for the affected nodes
+
+1. Determine the pod name for the Slingshot fabric manager pod and check the status of the fabric.
+
+   ```bash
+   ncn# kubectl exec -it -n services \
+     $(kubectl get pods --all-namespaces |grep slingshot | awk '{print $2}') \
+     -- fmn_status
+   ```
+
+#### Check for duplicate IP address entries
+
+1. Check for duplicate IP address entries in the Hardware State Management Database (HSM). Duplicate entries will cause DNS operations to fail.
+
+   1. Verify each node hostname resolves to one IP address.
+
+      ```bash
+      ncn# nslookup x9000c3s0b0n1
+      ```
+
+      Example output with one IP address resolving:
+
+      ```text
+      Server:         10.92.100.225
+      Address:        10.92.100.225#53
+
+      Name:   x9000c3s0b0n1
+      Address: 10.100.0.26
+      ```
+
+   1. Reload the KEA configuration.
+
+      ```bash
+      ncn# curl -s -k -H "Authorization: Bearer ${TOKEN}" -X POST -H "Content-Type: application/json" \
+                -d '{ "command": "config-reload",  "service": [ "dhcp4" ] }' https://api-gw-service-nmn.local/apis/dhcp-kea | jq
+      ```
+
+      If there are no duplicate IP addresses within HSM, the following response is expected:
+
+      ```json
+      [
+        {
+          "result": 0,
+          "text": "Configuration successful."
+        }
+      ]
+      ```
+
+      If there is a duplicate IP address, then an error message similar to the message below is expected. This example message indicates a duplicate IP address (`10.100.0.105`) in the HSM:
+
+      ```json
+      [{'result': 1, 'text': "Config reload failed: configuration error using file '/usr/local/kea/cray-dhcp-kea-dhcp4.conf': failed to add new host using the HW address '00:40:a6:83:50:a4 and DUID '(null)' to the IPv4 subnet id '0' for the address 10.100.0.105: There's already a reservation for this address"}]
+      ```
+
+1. Use the following example `curl` command to check for active DHCP leases.
```

Comment:
Does this step really belong in the middle of the "remove duplicate IP addresses" section? It just struck me as weird that the previous steps and the following step are all centered around determining if there are duplicate IPs and removing them if there are, but this one is about Kea DHCP leases. If it IS meant to be in the midst of the procedure, I'd suggest adding some context to explain its connection.

Reply:
I'm not sure why it is in the middle, this part of the procedure was copied from `operations/node_management/Adding_a_Liquid-cooled_blade_to_a_System.md`. I agree it seems a bit out-of-order here. I'll move it below this section about checking HSM, to line 307. I will repeat this with the other instances of this issue in the other files as well.